Kvstore tests api change to run on K64F only

theamirocohen opened the pull request:
Change KVStore API tests to run only on K64F, these tests check the KVStore functionality without actually testing the board's storage, Thus they should produce the same results for each device they run on.
We don't see a point loading the CI with these tests if they all produce the same result regardless of the board.
The one test that does run on the actual board's storage is static_tests, which wasn't changed.
K64F was selected for no special technical reason but only because of it being available and convenient to use.

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

ciarmcom commented:
@theamirocohen, thank you for your changes.
@ARMmbed/mbed-os-storage @ARMmbed/mbed-os-maintainers please review.

0xc0170 commented:
> Change KVStore API tests to run only on K64F, these tests only check functionality.

Why only on one target ?

davidsaada commented:
> Why only on one target ?

The modified tests check the functionality of the various KVStore APIs. Most of them run in memory (over the Heap block device). Thus they should produce the same results for each device they run on (providing it has enough RAM). In other words, these tests check the KVStore functionality without actually testing the board's storage. The one test that does run on the actual board's storage is `static_tests`, which wasn't changed in this PR. We don't see a point loading the CI with these tests if they all produce the same result regardless of the board.

dannybenor commented:
The real and wide test for different boards is the test for the static API. Until now, this test was also not executed in many targets because the default configuration was for external memory, and not many boards have the right components in targets.json.
With the new configuration, we will get better coverage, because the default will be selected according to the defined components in targets.json. This is one of the breaking changes.

0xc0170 commented:
I would like to have @ARMmbed/mbed-os-test review. 

This feels like there should be unit tests rather? Testing only on one target does not feel right (why only one and why that one will  be questions we get)

OPpuolitaival reviewed (approved):
Not optimal solution but good to run even with one board. These tests seems to need real device and therefore cannot be replaced with unit testing.

jeromecoutant commented:
Target | Toolchain | TestSuite | TestCase | mbed-os-5.11.0
-- | -- | -- | -- | --
DISCO_F407VG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_F407VG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
DISCO_F429ZI | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_F429ZI | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
DISCO_F746NG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_F746NG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
DISCO_F769NI | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_F769NI | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
DISCO_L072CZ_LRWAN1 | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_L072CZ_LRWAN1 | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F070RB | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F070RB | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F072RB | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F072RB | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F303RE | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F303RE | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F401RE | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F401RE | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F410RB | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F410RB | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F411RE | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F411RE | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F412ZG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F412ZG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F413ZH | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F413ZH | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F429ZI | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F429ZI | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F439ZI | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F439ZI | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F446ZE | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F446ZE | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F746ZG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F746ZG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_F756ZG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_F756ZG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_L432KC | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_L432KC | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_L433RC_P | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_L433RC_P | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_L486RG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_L486RG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_L496ZG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_L496ZG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
NUCLEO_L4R5ZI | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
NUCLEO_L4R5ZI | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK
DISCO_L496AG | ARM | features-storage-tests-kvstore-general_tests_phase_1 | features-storage-tests-kvstore-general_tests_phase_1 | OK
DISCO_L496AG | ARM | features-storage-tests-kvstore-general_tests_phase_2 | features-storage-tests-kvstore-general_tests_phase_2 | OK

0xc0170 commented:
@theamirocohen as already asked, can you add to the commit message a reason for limiting to just one target and why it was chosen that specific target?

0xc0170 reviewed (changes requested):
As my comment above ^^

0xc0170 commented:
The only blocking thing here is the request above and @ARMmbed/mbed-os-storage team approval

theamirocohen committed:
Kvstore tests api change to run on K64F only

Change KVStore API tests to run only on K64F, these tests check the KVStore functionality without actually testing the board's storage,
Thus they should produce the same results for each device they run on.
K64F was selected for no special technical reason but only because of it being available and convenient to use.

theamirocohen commented:
> @theamirocohen as already asked, can you add to the commit message a reason for limiting to just one target and why it was chosen that specific target?

The commits were merged and the commit message was changed

jeromecoutant commented:
What about removing features/storage/kvstore/mbed_lib.json file,
and add the KVSTORE_ENABLED macro only in your unit test environment ?

davidsaada commented:
> 
> 
> What about removing features/storage/kvstore/mbed_lib.json file,
> and add the KVSTORE_ENABLED macro only in your unit test environment ?

Good idea. @theamirocohen Can you take care of it? It's indeed not required now as the tests only run on K64F.

Commit:
Remove kvstore's mbed_lib.json file (not relevant any more)

In addition, use the NOT_SUPPORTED directive to skip the tests.

davidsaada commented:
Removed KVStore's mbed_lib.json file.

jeromecoutant reviewed (commented):
Then patch in test files are not needed any more ...?

davidsaada commented:
> Then patch in test files are not needed any more ...?

Which patches?

0xc0170 commented:
CI started

cmonr commented:
Relaunching Travis CI.

Not sure why, but results didn't propigate...

offirko commented:
@ARMmbed/mbed-os-maintainers - Seems like it failed again to propagate: 
(can you examine and rerun?)
[greentea-test] Starting building: mbed-os-ci_greentea-test #751
[Pipeline] [greentea-test] echo
[greentea-test] Exited https://mbed-os.mbedcloudtesting.com/job/mbed-os-ci_greentea-test with result hudson.AbortException: mbed-os-ci_greentea-test #751 completed with status FAILURE (propagate: false to ignore)

0xc0170 commented:
Test restarted

0xc0170 commented:
I restarted also travis (it's not yet appearing to be pending here)

0xc0170 commented:
Reopening this PR to get travis reenabled (does not report back anything but it's green)

0xc0170 commented:
Travis green 💯

0xc0170 commented:
All green
